sdk/core/src/http_client.rs

```diff
@@ -20,90 +21,40 @@ pub fn new_http_client() -> std::sync::Arc<dyn HttpClient> {
 #[cfg_attr(target_arch = "wasm32", async_trait(?Send))]
 #[cfg_attr(not(target_arch = "wasm32"), async_trait)]
 pub trait HttpClient: Send + Sync + std::fmt::Debug {
-    /// Send out a request using `hyperium/http`'s types.
-    ///
-    /// This method is considered deprecated and should not be used in new code.
-    async fn execute_request(&self, request: Request<Bytes>) -> crate::Result<Response<Bytes>>;
-
     /// Send out a request using `azure_core`'s types.
     ///
-    /// This function will be the only one remaining in the trait as soon as the trait stabilizes.
-    /// It will be renamed to `execute_request`. The other helper functions (ie
-    /// `execute_request_check_status`) will be removed since the status check will be
-    /// responsibility of another policy (not the transport one). It does not consume the request.
-    /// Implementors are expected to clone the necessary parts of the request and pass them to the
-    /// underlying transport.
-    async fn execute_request2(&self, request: &crate::Request) -> crate::Result<crate::Response>;
+    /// It does not consume the request. Implementors are expected to clone the necessary parts
+    /// of the request and pass them to the underlying transport.
+    async fn execute_request(&self, request: &crate::Request) -> crate::Result<crate::Response>;
 
-    /// Send out a request and validate it was in the `2xx` range, using
-    /// `hyperium/http`'s types.
-    ///
-    /// Note: the `expected_status` parameter is never used, and instead we
-    /// always validate the status was in the `2xx` range. This method should
```

Comment:
I have concerns regarding the implementation here.

While this comment says 2xx range, the original implementation checks 200..400 (which this PR matches) does not align with this comment.  More importantly, accepting a response within the 3xx range is probably not what the consumers of this API are likely to expect.

IMO, this is easily handled from the consumer side as:

`Response.status().is_success()`, or `Response.status() == HttpStatus::CREATED`